### ISSUE TITLE
UML Diagram Generation

### ISSUE THREAD

**Issue by ivan-pi** [lfortran/lfortran#4525]
A feature I'd find useful is to have a flag (or a separate tool) to generate UML diagrams from Fortran types and modules.

There are some examples of this in the literature:
- Extracting UML class diagrams from object-oriented fortran: ForUML, https://doi.org/10.1155/2015/421816
- Extended ForUML for Automatic Generation of UML Sequence Diagrams from Object-Oriented Fortran, https://doi.org/10.1155/2019/2542686

More generally, for other programming languages, you can find dozens of tools like this (it appears to be popular in CS programs), but the tools rarely find their way back into production software (obviously, there is a maintenance cost to support features like these).

The output could be a custom XML schema, or a format compatible with tools like [Mermaid](https://mermaid.js.org/) or [PlantUML](https://plantuml.com/), so it can be inserted into websites/documentation.

**Comment by certik** [comment]
Yes, this is a commonly requested feature. The compiler should be able to easily do it.
